server/plugin/api.go

```diff
@@ -1258,23 +1468,293 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
 	p.writeJSON(w, resp)
 }
 
+func (p *Plugin) updatePost(post *model.Post, issue *UpdateIssueRequest, w http.ResponseWriter) {
+	post, appErr := p.API.GetPost(issue.PostID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s", issue.PostID), StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to load the post %s : not found", issue.PostID), StatusCode: http.StatusNotFound})
+		return
+	}
+
+	post.Props[AssigneesForProps] = issue.Assignees
+	post.Props[LabelsForProps] = issue.Labels
+	post.Props[DescriptionForProps] = issue.Body
+	post.Props[TitleForProps] = issue.Title
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: fmt.Sprintf("failed to update the post %s", issue.PostID), StatusCode: http.StatusInternalServerError})
+	}
+}
+
+func (p *Plugin) validateIssueRequestToUpdate(issue *UpdateIssueRequest, w http.ResponseWriter) bool {
```

Comment:
```suggestion
func (p *Plugin) validateIssueRequestForUpdation(issue *UpdateIssueRequest, w http.ResponseWriter) bool {
```
This function should be in `utils`. Move all non-API functions to `utils`.